### PR TITLE
[ISSUE #7858]Correct the exception log in ServerHttpAgent

### PR DESCRIPTION
**Pull request by liqipeng**
## What is the purpose of the change

Correct the exception log in ServerHttpAgent

## Brief changelog

Correct the exception log in ServerHttpAgent

## Verifying this change
After this change, the log will be like below: 
```
[NACOS ConnectException httpPost] currentServerAddr: http://127.0.0.1:8848, err : Connection refused: connect
java.net.ConnectException: Connection refused: connect
	at java.net.DualStackPlainSocketImpl.waitForConnect(Native Method) ~[?:1.8.0_202]
	at java.net.DualStackPlainSocketImpl.socketConnect(DualStackPlainSocketImpl.java:85) ~[?:1.8.0_202]
	at java.net.AbstractPlainSocketImpl.doConnect(AbstractPlainSocketImpl.java:350) ~[?:1.8.0_202]
```

Follow this checklist to help us incorporate your contribution quickly and easily:

* [x] Make sure there is a Github issue filed for the change (usually before you start working on it). Trivial changes like typos do not require a Github issue. Your pull request should address just this issue, without pulling in other changes - one PR resolves one issue.
* [x] Format the pull request title like `[ISSUE #123] Fix UnknownException when host config not exist`. Each commit in the pull request should have a meaningful subject line and body.
* [x] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
* [ ] Write necessary unit-test to verify your logic correction, more mock a little better when cross module dependency exist. If the new feature or significant change is committed, please remember to add integration-test in [test module](https://github.com/alibaba/nacos/tree/master/test).
* [x] Run `mvn -B clean package apache-rat:check findbugs:findbugs -Dmaven.test.skip=true` to make sure basic checks pass. Run `mvn clean install -DskipITs` to make sure unit-test pass. Run `mvn clean test-compile failsafe:integration-test`  to make sure integration-test pass.

### CODE REVIEW

File: client/src/main/java/com/alibaba/nacos/client/config/http/ServerHttpAgent.java [hunk 1]
```diff
@@ -99,24 +99,26 @@
...
+                            currentServerAddr, result.getCode());
                 } else {
                     // Update the currently available server addr
                     serverListMgr.updateCurrentServerAddr(currentServerAddr);
                     return result;
                 }
             } catch (ConnectException connectException) {
                 LOGGER.error("[NACOS ConnectException httpGet] currentServerAddr:{}, err : {}",
-                        serverListMgr.getCurrentServerAddr(), connectException.getMessage());
+                        currentServerAddr, connectException.getMessage(), connectException);
```
Comment (KomachiSion) [inline client/src/main/java/com/alibaba/nacos/client/config/http/ServerHttpAgent.java:110]: For connection and sock connection， we don't want print much stack.

File: client/src/main/java/com/alibaba/nacos/client/config/http/ServerHttpAgent.java [hunk 2]
```diff
@@ -130,8 +132,8 @@
             
         } while (System.currentTimeMillis() <= endTime);
         
-        LOGGER.error("no available server");
-        throw new ConnectException("no available server");
+        LOGGER.error("Request time out or no available server, lastErrorSeverAddr: {}", lastErrorSeverAddr);
+        throw new ConnectException("Request time out or no available server, lastErrorSeverAddr: " + lastErrorSeverAddr);
```
Comment (KomachiSion) [inline client/src/main/java/com/alibaba/nacos/client/config/http/ServerHttpAgent.java:136]: The last error server is no useful. When code run in there, means all server nodes has retried. only print last one can't help know the problem.

**Review by KomachiSion — CHANGES_REQUESTED** [review]

**Review by KomachiSion — CHANGES_REQUESTED** [review]
After review, I think this PR has no need to merge, main changes is no meaningful.

`serverListMgr.getCurrentServerAddr()` only execute once both before and after.

So I think the pr should be close.

File: client/src/main/java/com/alibaba/nacos/client/config/http/ServerHttpAgent.java (hunk 1 above)
Comment (liqipeng) [reply client/src/main/java/com/alibaba/nacos/client/config/http/ServerHttpAgent.java:110]: Ok, maybe code in ```httpDelete``` should be optimized.
```
    @Override
    public HttpRestResult<String> httpDelete ... ...
            ... ...
            } catch (ConnectException connectException) {
                LOGGER.error("[NACOS ConnectException httpDelete] currentServerAddr:{}, err : {}",
                        serverListMgr.getCurrentServerAddr(), ExceptionUtil.getStackTrace(connectException));  // HERE !!!!!
            } catch (SocketTimeoutException stoe) {
                LOGGER.error("[NACOS SocketTimeoutException httpDelete] currentServerAddr:{}， err : {}",
                        serverListMgr.getCurrentServerAddr(), ExceptionUtil.getStackTrace(stoe));   // HERE !!!!!
            ... ...
```

File: client/src/main/java/com/alibaba/nacos/client/config/http/ServerHttpAgent.java (hunk 2 above)
Comment (liqipeng) [reply client/src/main/java/com/alibaba/nacos/client/config/http/ServerHttpAgent.java:136]: Please check my comment below: 
```
    @Override
    public HttpRestResult<String> httpGet(String path, Map<String, String> headers, Map<String, String> paramValues,
            String encode, long readTimeoutMs) throws Exception {
        final long endTime = System.currentTimeMillis() + readTimeoutMs; // !!!!!
        injectSecurityInfo(paramValues);
        String currentServerAddr = serverListMgr.getCurrentServerAddr();
        int maxRetry = this.maxRetry;
        HttpClientConfig httpConfig = HttpClientConfig.builder()
                .setReadTimeOutMillis(Long.valueOf(readTimeoutMs).intValue())   // set read timeout here
                .setConTimeOutMillis(ConfigHttpClientManager.getInstance().getConnectTimeoutOrDefault(100)).build();
        do {
            ... ...
                HttpRestResult<String> result = NACOS_RESTTEMPLATE
                        .get(getUrl(currentServerAddr, path), httpConfig, newHeaders, query, String.class);
            ... ...
            } catch (ConnectException connectException) {
                LOGGER.error("[NACOS ConnectException httpGet] currentServerAddr:{}, err : {}",
                        serverListMgr.getCurrentServerAddr(), connectException.getMessage());
            } catch (SocketTimeoutException socketTimeoutException) {
                LOGGER.error("[NACOS SocketTimeoutException httpGet] currentServerAddr:{}， err : {}",
                        serverListMgr.getCurrentServerAddr(), socketTimeoutException.getMessage());  // if request read timeout, exception will be here and  method ```NACOS_RESTTEMPLATE.get``` will take at least ```readTimeoutMs``` ms.
            } catch (Exception ex) {

        } while (System.currentTimeMillis() <= endTime); // And this condition will fail with a high probability.
        
        LOGGER.error("no available server"); // So the error message is not exactly in most cases. It's a misleading message.
        throw new ConnectException("no available server");
    }

```

Comment (liqipeng) [reply client/src/main/java/com/alibaba/nacos/client/config/http/ServerHttpAgent.java:136]: And what said is not ture, and maybe it's just a timeout case.
> When code run in there, means all server nodes has retried

**Comment by liqipeng** [comment]
Maybe I should repost the description to here:  ( https://github.com/alibaba/nacos/issues/7858 )


Describe the bug
1. If the currentServerAddr changed, the server address output in log will not be same with current request and it is not correctly.
    ```
    public HttpRestResult<String> httpGet ......
            ......
            } catch (ConnectException connectException) {
                LOGGER.error("[NACOS ConnectException httpGet] currentServerAddr:{}, err : {}",
                        serverListMgr.getCurrentServerAddr(), connectException.getMessage());
            } catch (SocketTimeoutException socketTimeoutException) {
                LOGGER.error("[NACOS SocketTimeoutException httpGet] currentServerAddr:{}， err : {}",
                        serverListMgr.getCurrentServerAddr(), socketTimeoutException.getMessage());
            } catch (Exception ex) {
                LOGGER.error("[NACOS Exception httpGet] currentServerAddr: " + serverListMgr.getCurrentServerAddr(),
                        ex);
                throw ex;
            }
     ```
2. The log or exception thrown in the end of httpGet method output a message not covered all exception conditions. Maybe timeout.
    ```
    public HttpRestResult<String> httpGet ......
        ......
        LOGGER.error("no available server");
        throw new ConnectException("no available server");
   ```

**Pull request closed without merging**